scripts/templates/detector_vars.tf.j2

```diff
@@ -0,0 +1,54 @@
+{%- set id = name | replace(' ', '_') -%}
+{%- set default_aggregation = aggregation | default(true) -%}
+{%- set default_transformation = transformation | default(true) -%}
+# {{ id }} detector
+
+variable "{{ id }}_notifications" {
+  description = "Notification recipients list per severity overridden for {{ id }} detector"
+  type        = map(list(string))
+  default     = {}
+}
+
+{% if default_aggregation -%}
+variable "{{ id }}_aggregation_function" {
+  description = "Aggregation function and group by for {{ id }} detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = "{% if default_aggregation != true %}{{ aggregation | default("") }}{% endif %}"
+}
+
+{% endif -%}
+
+{% if default_transformation -%}
+variable "{{ id }}_transformation_function" {
+  description = "Transformation function for {{ id }} detector (i.e. \".mean(over='5m')\")"
+  type        = string
+  default     = "{% if default_transformation != true %}{{ transformation | default("") }}{% endif %}"
+}
+
+{% endif -%}
+
+variable "{{ id }}_disabled" {
+  description = "Disable all alerting rules for {{ id }} detector"
+  type        = bool
+  default     = null
+}
+
+{% for key, rule in rules.items() -%}
+  {%- set severity = key | lower -%}
+variable "{{ id }}_disabled_{{ severity }}" {
+  description = "Disable {{ severity }} alerting rule for {{ id }} detector"
+  type        = bool
+  default     = null
+}
+
+{% endfor -%}
+
+{% for key, rule in rules.items() -%}
+  {%- set severity = key | lower -%}
+variable "{{ id }}_threshold_{{ severity }}" {
+  description = "{{ severity | capitalize }} threshold for {{ id }} detector"
+  type        = number
+  default     = {{ rule.threshold }}
```

Comment:
if threshold is not specified, j2 put "None" here, what about a if instead? (so there is no default generated)

Reply:
yeah good catch, I think we will probably enrich the templates as things progress depending of users recurrent feedback.